### PR TITLE
refactor: centralized state management, performance optimizations, and vim-style navigation

### PR DESCRIPTION
**Pull request by nh13**
## Summary

Major refactoring effort covering architecture, performance, and UX improvements:

### State Management & Architecture
- Convert TUI to package structure (`tui/` with focused modules)
- Split parser into focused modules (`line_parser`, `log_reader`, `metadata`, `progress_tracker`)
- Add validation framework with `EventAccumulator` for cross-source verification
- Centralize constants, exceptions, and type definitions
- Add `ClockUtils` for testable time handling

### Performance Optimizations
- Replace `rglob` with `os.scandir` for 6-7x faster directory iteration
- Add `MetadataCache` for mtime-based caching of parsed metadata files
- Add `__slots__` to hot-path dataclasses (`JobInfo`, `TimeEstimate`, `SnakeseeEvent`, `MetadataRecord`)
- Add parallel file reading for large directories (>1000 files)
- Use `orjson` for faster JSON parsing
- Add fast-path prefix checks in log parsing

### Vim-Style Navigation
- Add two-mode navigation: Table Mode → Log Mode
- Table Mode: `j/k` rows, `g/G` first/last, `Ctrl-d/u` half-page, `Ctrl-f/b` full-page
- Log Mode: `j/k` scroll, `g/G` start/end, `Ctrl-d/u/f/b` page scroll
- `h/l` to switch between running/completions tables
- Change refresh rate keys from `h/j/k/l` to `+/-`, `</>`, `0`, `G`

### Testing
- Add 11 new benchmark tests for performance tracking
- Add property-based tests with Hypothesis
- Add comprehensive TUI keyboard edge case tests
- Add parser edge case tests
- Achieve 80%+ test coverage

## Test plan
- [x] All 914 tests pass
- [x] Coverage at 80.38%
- [x] Benchmarks show expected performance improvements
- [x] Manual testing of vim-style navigation

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Expanded TUI navigation with modal table/log modes, new shortcuts (q, ?, p, e, w, r, Ctrl+r) and runtime wildcard toggle; improved ETAs and history-aware estimates.

* **Bug Fixes**
  * More robust event/log handling, thread-safe event emission, improved error messages, and atomic profile save behavior.

* **Documentation**
  * README reorganized; added architecture and usage guides covering shortcuts and conditioning.

* **Performance**
  * Faster, cached metadata iteration and parser/estimator optimizations.

* **Tests**
  * Large new unit and benchmark suites across parser, estimator, plugins, state, and TUI.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Comment by nh13** [comment]
@coderabbitai review

File: snakesee/state/workflow_state.py [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+            self.status = WorkflowStatus.COMPLETED
+        elif self.completed_count == 0 and self.running_count == 0:
+            self.status = WorkflowStatus.NOT_STARTED
+        else:
+            # Check for stale workflow
+            if self.elapsed_seconds is not None:
+                if self.elapsed_seconds > self.config.time.stale_workflow_threshold:
+                    self.status = WorkflowStatus.STALE
+                else:
+                    self.status = WorkflowStatus.RUNNING
```
[the remark this answers is not in the document]
Comment (nh13) [reply snakesee/state/workflow_state.py]: ✅ Addressed in commit f1af558 - Fixed update_status logic to handle failed jobs when total_jobs is None and removed redundant (total_jobs or 0) check.

File: snakesee/constants.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+#: Maximum refresh rate in seconds (slowest)
+MAX_REFRESH_RATE: float = REFRESH_RATE_CONFIG.max_rate
+
+#: Default refresh rate in seconds
+DEFAULT_REFRESH_RATE: float = REFRESH_RATE_CONFIG.default_rate
+
+#: Seconds since last log modification before considering workflow stale/dead
+#: Default is 30 minutes (1800 seconds)
+#: SYNC: This value must match TimeConstants.stale_workflow_threshold in state/config.py
+STALE_WORKFLOW_THRESHOLD_SECONDS: float = 1800.0
```
[the remark this answers is not in the document]
Comment (nh13) [reply snakesee/constants.py]: ✅ Addressed in commit f1af558 - Updated comment to clarify that state/config.py imports STALE_WORKFLOW_THRESHOLD_SECONDS from this module (single source of truth).

File: snakesee/estimation/estimator.py [hunk 3]
```diff
@@ -0,0 +1,893 @@
...
+                        else (rule_mean * self.config.variance.rule_fallback) ** 2
+                    )
+                else:
+                    # Unknown rule: use global mean with higher uncertainty
+                    rule_mean = global_mean
+                    rule_var = (global_mean * self.config.variance.global_fallback) ** 2
+
+                total_expected += count * rule_mean
+                total_variance += count * rule_var
+
```
[the remark this answers is not in the document]
Comment (nh13) [reply snakesee/estimation/estimator.py]: ✅ Addressed in commit f1af558 - Changed pending count calculation to use progress.pending_jobs which correctly excludes failed and incomplete jobs.

File: snakesee/parser/line_parser.py [hunk 4]
```diff
@@ -0,0 +1,288 @@
...
+    """
+    try:
+        result = int(value)
+        if result <= 0:
+            logger.debug("Invalid %s: %d (must be positive)", field_name, result)
+            return None
+        return result
+    except ValueError:
+        logger.debug("Could not parse %s as integer: %s", field_name, value)
+        return None
```
[the remark this answers is not in the document]
Comment (nh13) [reply snakesee/parser/line_parser.py]: ✅ Addressed in commit f1af558 - Removed duplicate helper functions (_parse_timestamp, _parse_wildcards, _parse_positive_int) and imported them from snakesee/parser/utils.py instead.
